File: playgrounds/notification-service-gradle-sb-2/src/main/kotlin/com/sivalabs/ft/notifications/testdata/AirportTestDataQueryRunner.java
package org.springframework.samples.petclinic.testdata.airport;

import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.event.EventListener;
import org.springframework.data.domain.PageRequest;
import org.springframework.stereotype.Component;

@Component
public class AirportTestDataQueryRunner {

	private final AirportTestDataService airportTestDataService;

	public AirportTestDataQueryRunner(AirportTestDataService airportTestDataService) {
		this.airportTestDataService = airportTestDataService;
	}

	@EventListener(ApplicationReadyEvent.class)
	public void runTests() {
		airportTestDataService.runNplusOne_1();

		PageRequest pageRequest = PageRequest.of(0, 2);
		airportTestDataService.runPagination_1(pageRequest);
	}
}

File: playgrounds/notification-service-gradle-sb-2/src/main/kotlin/com/sivalabs/ft/notifications/testdata/AirportTestDataQueryRunner.kt
package com.sivalabs.ft.notifications.testdata

import org.springframework.context.annotation.Profile
import org.springframework.data.domain.PageRequest
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
@Profile("default", "local")
@EnableScheduling
class AirportTestDataQueryRunner(
    private val airportTestDataService: AirportTestDataService,
) {
    @Scheduled(initialDelay = 10000, fixedRate = 60000)
    fun runTests() {
        airportTestDataService.runNplusOne1()

        val pageRequest = PageRequest.of(0, 2)
        airportTestDataService.runPagination1(pageRequest)
    }
}
